lambdas/bearer-token-handler/src/bearer-token-handler.ts

```diff
@@ -39,8 +39,15 @@ export class BearerTokenHandler implements LambdaInterface {
         };
       }
 
+      let responseBody;
+      try {
+        responseBody = JSON.stringify(await response.json());
+      } catch (ignored) {
+        /* empty */
+      }
+
       throw new Error(
-        `Error response received from HMRC ${response.status} ${response.statusText}`
+        `Error response received from HMRC ${response.status} ${response.statusText}: ${responseBody}`
```

Comment:
I wondered about this before - are we sure that it's safe to expose this or is there a possibility it could contain something sensitive?

Reply:
Yeah I understand that we have everything in place from our side, I'm just worried about logging a response that is returned by HMRC as it's not something that we have control over and this wouldn't be picked up by our redactor.

My assumption (and what we've seen in testing so far) is that they wouldn't put sensitive information in a response body but we have seen that their responses don't always follow best practises which makes me a little nervous

Reply:
I have updated this PR to a draft and then if we decide to go ahead with it, the work is still here.